### PR TITLE
Improve versioning

### PR DESCRIPTION
**Pull request by oeway**
read version number from a json file, and add badge for showing engine version and api_version

### CODE REVIEW

File: imjoy/imjoyPluginEngine.py [hunk 1]
```diff
@@ -62,10 +63,16 @@ def get_drives():
 except ImportError:
     from queue import Queue, Empty  # python 3.x
 
+# read version information from file
+HERE = pathlib.Path(__file__).parent
+version_info = json.loads((HERE / ".." / "VERSION").read_text())
+
+__version__ = version_info["version"]
+__api_version__ = version_info["api_version"]
```
Comment (MartinHjelmare) [inline imjoy/imjoyPluginEngine.py]: If `__api_version__` is only intended to be used within this file, it should just be a constant, I think.

File: imjoy/imjoyPluginEngine.py [hunk 2]
```diff
@@ -1019,9 +1026,9 @@ def exited(result):
 
         logger.info("register client: %s", kwargs)
 
-        engine_info = {"api_version": API_VERSION}
-        if engine_version is not None:
-            engine_info["version"] = engine_version
+        engine_info = {"api_version": __api_version__}
+        if __version__ is not None:
```
Comment (MartinHjelmare) [inline imjoy/imjoyPluginEngine.py]: When can this not happen?

Comment (oeway) [reply imjoy/imjoyPluginEngine.py]: For got to remove

File: imjoy/imjoyPluginEngine.py (hunk 1 above)
Comment (oeway) [reply imjoy/imjoyPluginEngine.py]: I was trying to make it looks like __version__, so I should rename it as `API_VERSION`?

Comment (MartinHjelmare) [reply imjoy/imjoyPluginEngine.py]: Yes, I think so, unless we have a specific use case for the magic module attribute that requires it to be that.

**Pull request merged**